File: backend/app/schemas/cliente.py
from pydantic import BaseModel, EmailStr, Field
from datetime import datetime
from typing import Optional


class ClienteCreate(BaseModel):
    numero_cliente: str = Field(..., max_length=20)
    tipo: str = Field(..., max_length=10)
    nombre: str = Field(..., min_length=1, max_length=255)
    apellido_paterno: Optional[str] = Field(None, max_length=100)
    apellido_materno: Optional[str] = Field(None, max_length=100)
    razon_social: Optional[str] = Field(None, max_length=255)
    rfc: Optional[str] = Field(None, max_length=13)
    email: EmailStr
    telefono: Optional[str] = Field(None, max_length=20)
    telefono_alternativo: Optional[str] = Field(None, max_length=20)
    direccion: Optional[str] = None
    ciudad: Optional[str] = None
    estado: Optional[str] = None
    codigo_postal: Optional[str] = None
    notas: Optional[str] = None


class ClienteUpdate(BaseModel):
    tipo: Optional[str] = None
    nombre: Optional[str] = Field(None, min_length=1, max_length=255)
    apellido_paterno: Optional[str] = None
    apellido_materno: Optional[str] = None
    razon_social: Optional[str] = None
    rfc: Optional[str] = None
    email: Optional[EmailStr] = None
    telefono: Optional[str] = None
    telefono_alternativo: Optional[str] = None
    direccion: Optional[str] = None
    ciudad: Optional[str] = None
    estado: Optional[str] = None
    codigo_postal: Optional[str] = None
    notas: Optional[str] = None


class ContactoClienteCreate(BaseModel):
    nombre: str = Field(..., min_length=1, max_length=255)
    puesto: Optional[str] = None
    email: Optional[str] = None
    telefono: Optional[str] = None
    es_principal: bool = False


class ContactoClienteResponse(BaseModel):
    id: str
    nombre: str
    puesto: Optional[str] = None
    email: Optional[str] = None
    telefono: Optional[str] = None
    es_principal: int
    activo: bool

    model_config = {"from_attributes": True}


class ClienteResponse(BaseModel):
    id: str
    numero_cliente: str
    tipo: str
    nombre: str
    apellido_paterno: Optional[str] = None
    apellido_materno: Optional[str] = None
    razon_social: Optional[str] = None
    rfc: Optional[str] = None
    email: str
    telefono: Optional[str] = None
    telefono_alternativo: Optional[str] = None
    direccion: Optional[str] = None
    ciudad: Optional[str] = None
    estado: Optional[str] = None
    codigo_postal: Optional[str] = None
    notas: Optional[str] = None
    activo: bool
    created_at: datetime
    updated_at: datetime
    contactos: list[ContactoClienteResponse] = []

    model_config = {"from_attributes": True}


class ClienteListResponse(BaseModel):
    items: list[ClienteResponse]
    total: int
    skip: int
    limit: int

File: frontend/src/types/cliente.ts
export type TipoCliente = 'persona_fisica' | 'persona_moral' | 'gobierno'
export type EstadoCuenta = 'al_dia' | 'debe' | 'credito'
export type FormaPago = 'transferencia' | 'cheque' | 'efectivo' | 'tarjeta'

export interface Cliente {
  id: string
  tipo: TipoCliente
  nombre: string
  rfc: string | null
  email: string
  telefono: string | null
  telefono_alternativo: string | null
  direccion: string | null
  ciudad: string | null
  estado_geo: string | null
  codigo_postal: string | null
  fecha_nacimiento: string | null
  representante_legal: string | null
  giro_empresarial: string | null
  calificacion: number
  estado_cuenta: EstadoCuenta
  limite_credito: number
  descuento_especial: number
  forma_pago_preferida: FormaPago | null
  notas: string | null
  activo: boolean
  created_at: string
  updated_at: string
  contactos: ContactoCliente[]
}

export interface ContactoCliente {
  id: string
  nombre: string
  puesto: string | null
  email: string | null
  telefono: string | null
  es_principal: number
  activo: boolean
}

export interface ClienteCreate {
  tipo: TipoCliente
  nombre: string
  rfc?: string
  email: string
  telefono?: string
  telefono_alternativo?: string
  direccion?: string
  ciudad?: string
  estado_geo?: string
  codigo_postal?: string
  fecha_nacimiento?: string
  representante_legal?: string
  giro_empresarial?: string
  calificacion?: number
  forma_pago_preferida?: FormaPago
  notas?: string
}

export interface ClienteUpdate extends Partial<ClienteCreate> {
  estado_cuenta?: EstadoCuenta
  limite_credito?: number
  descuento_especial?: number
}
